index.ts

```diff
@@ -196,26 +196,29 @@ function suiteClassCallback(target: SuiteCtor, context: TestFunctions) {
 
             // assert testName is truthy
             if (parameters) {
-                const nameForParameters = method[nameForParametersSymbol];
-                parameters.forEach((parameterOptions, i) => {
-                    const { mark, name, params } = parameterOptions;
-
-                    let parametersTestName = `${testName}_${i}`;
-                    if (name) {
-                        parametersTestName = name;
-                    } else if (nameForParameters) {
-                        parametersTestName = nameForParameters(params);
-                    }
-
-                    const shouldSkipParam = shouldSkip || (mark === Mark.skip);
-                    const shouldOnlyParam = shouldOnly || (mark === Mark.only);
-                    const shouldPendingParam = shouldPending || (mark === Mark.pending);
-
-                    const testFunc = ((shouldPendingParam || shouldSkipParam) && context.it.skip)
-                            || (shouldOnlyParam && context.it.only)
-                            || context.it;
-
-                    applyTestFunc(testFunc, parametersTestName, method, [params]);
+                // we make the parameterised test a child suite so we can late bind the parameterised tests
+                context.describe(testName, function() {
+                    const nameForParameters = method[nameForParametersSymbol];
+                    parameters.forEach((parameterOptions, i) => {
+                        const { mark, name, params } = parameterOptions;
```

Comment:
Now it seems you only need to check if these parameters are a generator and iterate it if it is. :)

Reply:
This is not part of this PR. See #144

Reply:
This is just in preparation for that.